### ISSUE TITLE
Can not Activate Mageplaza extensions

### ISSUE THREAD

**Issue by imsamthomas** [mageplaza/module-core#26]
![image](https://user-images.githubusercontent.com/2810172/57203040-633ae780-6fd6-11e9-86e3-41036fa8e7ad.png)

## Problems:

You may be asked for Activating your extension but got the following messages:

### Problem 1: 

```
A technical problem with the server created an error. Try again to continue what you were doing. If the problem persists, try again later.
```

#### Problem 2: No message
There is no message appear except for console log:

```
{"success":false}
```

#### Problem 3: Cannot connect to server

Error Message:

```
Cannot connect to server. Please try again later.
```

## Solutions

You should upgrade [mageplaza core](https://github.com/mageplaza/module-core) module to version 1.4.3 or greater.

**Update to v1.4.3**
```
composer update mageplaza/module-core:1.4.3
```

**Update to Latest version**
```
composer update mageplaza/module-core
```

**Comment by haitv282** [comment]
If you cannot update to Module core v1.4.3, try this:

```
composer remove mageplaza/module-core
composer require mageplaza/module-core
```

Your store will be updated to v1.4.3

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 1 update, 0 removals
  - Updating mageplaza/module-core (1.4.2 => 1.4.3): Downloading (100%
```

**Comment by ywsw** [comment]
I am still having an issue trying to validate Mageplaza SMTP 1.2.10 (latest version). I am using the latest version of Mageplaza core (1.4.3) on Magento 2.3.1

When I click on the "Activate Now" button, I get an error message saying "Error!"

**Comment by Jaki117** [comment]
I have the same Problem.
I am using Magento 2.2.3. I have Mageplaza SMTP 1.2.10 with Mageplaza Core 1.4.3 installed.
When I try to activate I get following message after 1 minute loading:
A technical problem with the server created an error. Try again to continue what you were doing. If the problem persists, try again later.
I already tried reinstalling Magento and the extension.

**Comment by rajeshaspl** [comment]
When you get the account not activate error, here is the solution:
Run below INSERT query in database and replace the CAPITAL LETTERS INFORMATION:

1) ENTER YOUR LICENSE KEY HERE FROM MAGE PLAZA EXTENSION SITE'
2) YOUR EMAIL ADDRESS
3) YOUR NAME

INSERT INTO `core_config_data` (`config_id`, `scope`, `scope_id`, `path`, `value`) VALUES
(NULL, 'default', 0, 'mpsocialshare/module/active', '1'),
(NULL, 'default', 0, 'mpsocialshare/module/product_key', 'ENTER YOUR LICENSE KEY HERE FROM MAGEPLAZA EXTENSION SITE'),
(NULL, 'default', 0, 'mpsocialshare/module/email', 'YOUR EMAIL ADDRESS'),
(NULL, 'default', 0, 'mpsocialshare/module/name', 'YOUR NAME'),
(NULL, 'default', 0, 'mpsocialshare/module/create', '1'),
(NULL, 'default', 0, 'mpsocialshare/module/subscribe', '1'),
(NULL, 'default', 0, 'mpsocialshare/float/select_page', NULL),
(NULL, 'default', 0, 'mpsocialshare/float/cms_page', NULL);

Clear the magento cache and check admin site it's activate now.

Thanks
Rajesh

**Comment by baoutch** [comment]
Here is a quick and very dirty solultion
In vendor/mageplaza/module-core/Controller/Adminhtml/Index/Activate.php  
For instance for blog extension I just grabbed my key on Mageplaza customer account and set it manually 
```
if($params['extension'] == 'Mageplaza_Blog'){
    $result['success'] = true;
    $result['key'] = 'BLOG-XXXXXXXXXXXXXXXX';
}
```
I know this is dirty but it fixes until I find the time to upgrade test and stuff and allows to use the modules.

**Comment by mmicciolo** [comment]
Still the same issues on Magento 2.3.1 and Mageplaza-Core 1.4.3.

Trying to use Mageplaza SMTP 1.2.10.

**Comment by rajeshaspl** [comment]
@mmicciolo , Hi Provide the your ssh details I can solve for you

**Comment by imsamthomas** [comment]
> @mmicciolo , Hi Provide the your ssh details I can solve for you
> contact at rajesh.ramvani at gmail

I updated the solution, please check https://github.com/mageplaza/module-core/issues/26#issue-440512550

**Comment by mmicciolo** [comment]
@imsamthomas 

Upgraded to 1.4.3 no issues. Problem is a I am still facing "A technical problem with the server created an error. Try again to continue what you were doing. If the problem persists, try again later." when trying to activate the free SMTP module.

**Comment by HuynhThien55310** [comment]
Same problem with the latest version 1.4.4

**Comment by GKnd** [comment]
Same problem with latest version for me too

**Comment by HuynhThien55310** [comment]
I found a temporary solution. You can go to your_module/etc/config.xml 
`<module>
                <type>1</type>
 </module>`
change 1 to 0. The extension will be actived.

**Comment by reense** [comment]
I am having the same issue as described above.

**Comment by ktsixit** [comment]
I have the same issue. Core extension is updated to 1.4.3 but I canot active extension. I also tried
https://github.com/mageplaza/module-core/issues/26#issuecomment-491701716 with no luck.
I have used this extension in the past without any issues. This is the second attempt to install in the past days and it fails. There must be some kind of issue here.

**Comment by HuynhThien55310** [comment]
I found the solution and it works. Also, I posted it to here but Mageplaza deleted my comment. I think my solution effects the license of extension. So they deleted my comment. I just want to help people and I really appreciate what Mageplaza do, I don't want to harm their business. If anyone really needs a solution, feel free to email me at "re.orionbelt@gmail.com". Hope they fix this issue as soon as possible.

**Comment by imsamthomas** [comment]
Hi @reense , @ktsixit 

We have updated **Solution 3**, please check.

**Comment by MihaiTudorRO** [comment]
> mpsocialshare

For other modules like blog and smtp just replace on each line mpsocialshare with your extension name . ex. blog or smtp. it worked, and also do not forget to clean cache. Thanks @rajeshaspl .

**Comment by dnyt84** [comment]
Same problem here.
I am on core 1.4.4 and using same order number 1.0.1
When I ssh into the box I can 1) ping dashboard.mageplaza.com and it resolves correctly 2) wget dashboard.mageplaza.com
But in store it just sits there waiting and eventually I get "A technical problem with the server created an error. Try again to continue what you were doing. If the problem persists, try again later."
I did step 1 and 3 without any luck ... :/

**Comment by thiagogalvao** [comment]
I have the same issue that was described as @dnyt84 on post above.
I'm running the same version also.

**Comment by mesnitu** [comment]
Same problem here. Version 1.4.4.

**Comment by LiamKarlMitchell** [comment]
Thanks
Deleting the app/code/Mageplaza/core and running this
```
composer require "mageplaza/module-core:1.4.3" --ignore-platform-reqs
rm -rf generated/code/
```

Worked on a Development mode site I'm working on.

Turns out the previous core version was too old or something.
I installed the modules by copy pasting to the app folder, but the composer way seems better?

**Comment by wltprgm** [comment]
Not working

1.4.4 and 1.4.3

**Comment by shelebg** [comment]
I have the same problem. I tried everything and I did not solve the problem. Do not have /etc/hosts   how to configure it? where to add 198.199.79.159   dashboard.mageplaza.com?

**Comment by rajeshaspl** [comment]
> I have the same problem. I tried everything and I did not solve the problem. Do not have /etc/hosts   how to configure it? where to add 198.199.79.159 dashboard.mageplaza.com?

If you have linux os you can find hosts file /etc/hosts access through root or sudo user.
if you have windows system check system32/drivers/etc/hosts file

**Comment by shelebg** [comment]
thanks for your reply
even that did not solve the problem

**Comment by rajeshaspl** [comment]
First run below commands:

php -dmemory_limit=1G -f bin/magento setup:upgrade
after setup upgrade your new extension should be display in the list.

php -dmemory_limit=1G -f bin/magento setup:di:compile
php -d memory_limit=1G bin/magento setup:static-content:deploy -f

module name you cat get from app/code/MagaPlaza/Extensionfolder/etc/module.xml file

Like this Ex. <module name="Mageplaza_EmailAttachments" setup_version="1.0.0">

php -d memory_limit=1G bin/magento module:enable Module_Extension_Name

check the config.xml file under app/code/MagaPlaza/Extensionfolder/etc/config.xml

Ex : <mp_email_attachments> this is our module Config_extension_name, this entry require into database
search core_config_data data table find the entry available if not then insert like this

INSERT INTO `core_config_data` (`config_id`, `scope`, `scope_id`, `path`, `value`) VALUES
(NULL, 'default', 0, 'Config_extension_name/module/active', '1');

After insert above query again run commands, setup:upgrade, compile and deploy

**Comment by heyfletch** [comment]
I have all the same problems mentioned above. None of the solutions work on local or staging. 1.4.4. I'm just trying to use the SMTP extension, but I can't set it up because I can't activate the extension. stuck.

**Comment by HuynhThien55310** [comment]
I posted the solution 2 months ago. But they deleted it although it
actually works and really easy to do just because they won't receive your
email for some business by doing this. Can't believe this, i'm so sad.

**Comment by heyfletch** [comment]
What’s the solution?

**Comment by blueprintagencies** [comment]
I'm having the same issue with the smtp module I've tried all the suggestions but doesn't seem to work. I tried updating core_config_data table using mpsmtp in the path. I have the latest core, have cleared my cache and compiled but still not showing as activated. This is in Magento 2.2.7

**Comment by shelebg** [comment]
I put it through the composer
magento-2-blog-extension
magento-2-seo-extension
module-core
module-sitemap
and all of them have the same problem.
Whether this can be solved or the Mageplaza extensions should be avoided?

**Comment by HuynhThien55310** [comment]
Go to extension_you_want_to_active/etc/config.xml. 
`<module>`
              ` <type>1</type>`
  `</module>`
Change 1 to 0. The extension will be activated without contact with Mageplaza. This worked for me, hope this helps.

**Comment by heyfletch** [comment]
Yes, that worked. Thanks! Be sure to flush cache to take effect.

**Comment by heyfletch** [comment]
Do you know if there's a way to lock that config so I can commit it git, via something like magento config:set --lock-config?

**Comment by shelebg** [comment]
ХуинхТхиен55310  
Thank you very much !
SEO is working with me now!

**Comment by maxacarvalho** [comment]
I'm trying to use the SMTP extension, I tried all the solutions and nothing works. I wonder why you have this activation process for a free extension, simply because you want to send spam to the shops that are using it.
Make it paid instead, that's a lot better and more fair.

I'll simply fork the repository and remove the validation, a lot easier.

**Comment by secaliskan** [comment]
same here, magento 2.3.0 php 7.2 
/etc/hosts include the ip address mapping, mageplaza module core is 1.4.4 (greater than 1.4.3) and still no activation.

now other slider extension is also not working? 
What are you guys doing, tell us we pay for the extensions,
Why for free extension you have activation process which is not working.

total time consuming, we are developers not customers, we try to give the project on time to our customers... and with this its not possible...

**Comment by HuynhThien55310** [comment]
@secaliskan 
Have you tried my solution above?

**Comment by secaliskan** [comment]
@HuynhThien55310 
yes tried and worked like a chnarm, thanks for that... You are a superb magento developer...

yes what he advised is really working, now there's no activation information anymore. Looking like an extension without activation...

**Comment by bachchasingh** [comment]
Form SMTP, You can directly run the following query with value
**INSERT INTO core_config_data (config_id, scope, scope_id, path, value) VALUES (NULL, 'default', 0, 'smtp/module/active', '1');
INSERT INTO core_config_data (config_id, scope, scope_id, path, value) VALUES (NULL, 'default', 0, 'smtp/module/product_key', 'Your Key');
INSERT INTO core_config_data (config_id, scope, scope_id, path, value) VALUES (NULL, 'default', 0, 'smtp/module/email', 'Your Email');
INSERT INTO core_config_data (config_id, scope, scope_id, path, value) VALUES (NULL, 'default', 0, 'smtp/module/name', 'Your Name');
INSERT INTO core_config_data (config_id, scope, scope_id, path, value) VALUES (NULL, 'default', 0, 'smtp/module/create', '1');
INSERT INTO core_config_data (config_id, scope, scope_id, path, value) VALUES (NULL, 'default', 0, 'smtp/module/subscribe', '1');**

Clear the cache and in back-end it should be activate

**Comment by fmsthird** [comment]
updating the core from this https://github.com/mageplaza/module-core
works for me.

**Comment by rafaelpatro** [comment]
I've installed the latest Core release 1.4.9 and I'm facing the same problem.
![image](https://user-images.githubusercontent.com/13813964/92956553-71580e00-f43d-11ea-82ae-5aadb5bff81f.png)

mageplaza-core: 1.4.9
mageplaza-smtp: 1.4.3

**Comment by yourmedsin** [comment]
**Go to file:** /home/<username>/public_html/app/code/Mageplaza/Core/Model/Activate.php

**Changes:** Please change the MAGEPLAZA_ACTIVE_URL from 'http://store.mageplaza.com/license/index/activate' to 'https://dashboard.mageplaza.com/license/index/activate/?isAjax=true'

**Correct config:**
`
const MAGEPLAZA_ACTIVE_URL = 'https://dashboard.mageplaza.com/license/index/activate/?isAjax=true';

`
Refresh the Mageplaza configuration page and then Enter Registration name and email. Hit Return. It solves the issue.

<img width="1144" alt="Screenshot 2021-04-14 at 12 38 08 AM" src="https://user-images.githubusercontent.com/78869899/114607068-b5540f80-9cb9-11eb-8c83-d92c33af8c5e.png">

<img width="1087" alt="Screenshot 2021-04-14 at 12 36 26 AM" src="https://user-images.githubusercontent.com/78869899/114607007-a0777c00-9cb9-11eb-9994-c3ed5d29a038.png">

**Comment by nicti** [comment]
Using `magplaza/module-core` 1.4.10, we are still encountering this problem. The request sent to `https://dashboard.mageplaza.com/license/index/activate/?isAjax=true` returns 302 without and with Solution 3 applied.

**Comment by dro27** [comment]
using `mageplaza/module-core 1.4.11`. i encountered this `cannot connect to server. please try again later` message as well. solutions provided didnt work for me except those workarounds provided like [thiagogalvao](https://github.com/thiagogalvao). 

here is my other solutions for this. I went to take a look at the curl response `\Mageplaza\Core\Model\Activate.php` `$resultCurl` and saw it was empty. btw they're using this `Magento\Framework\HTTP\Adapter\CurlFactory`. I decided to change that to `Magento\Framework\HTTP\Client\Curl` [docs here](https://devdocs.magento.com/guides/v2.4/get-started/gs-curl.html). cuz i think maybe the class wasn't working properly.

I made the necessary changes and commented the original lines
I commented this lines:
```
$this->curlFactory->create()
$curl->write()
$curl->read()
$curl->close()
Zend_Http_Response::extractBody($resultCurl)
AbstractData::jsonDecode($responseBody)
```

and added this
```
$this->curl->post(self::MAGEPLAZA_ACTIVE_URL, $params);
$result = json_decode($this->curl->getBody(), true);
```
im not using ssl locally so i added this
```
$this->curl->setOption(CURLOPT_SSL_VERIFYHOST, 0);
$this->curl->setOption(CURLOPT_SSL_VERIFYPEER, 0);
```
and i just followed the variables used.

If anyone experienced the issue same as me, try to change the curl class mageplaza used at `\Mageplaza\Core\Model\Activate.php`. but of course remember that you made the changes before doing any update.

hope this helps anyone who came here for the solution

**Comment by karthickkg** [comment]
Hi,
Had the same activation issue with Mageplaza. The issue is actually due to changes with Php 8.1. Atleast this was the case with mine. 
In file _vendor/mageplaza/module-core/Model/Activate.php_
Change line _http_build_query($params, null, '&')_
To _http_build_query($params, ‘’, '&')_
Passing null to string argument is depreciated in php 8.1.

**Comment by themew** [comment]
> Hi, Had the same activation issue with Mageplaza. The issue is actually due to changes with Php 8.1. Atleast this was the case with mine. In file _vendor/mageplaza/module-core/Model/Activate.php_ Change line _http_build_query($params, null, '&')_ To _http_build_query($params, ‘’, '&')_ Passing null to string argument is depreciated in php 8.1.

Tried this method, and rather than the screen hanging, I get the rotating graphic and this error `A technical problem with the server created an error. Try again to continue what you were doing. If the problem persists, try again later.`

This is using smtp version 4.7.1 with php 8.1 on M2.4.4  :(

**Comment by cbtrident** [comment]
so what the latest on this as am having the same problem am php 7.4, smtp 4.7.3 core 1.4.12
as i still getting Cannot connect to server. Please try again later. i have add the hosts to the file
but after this not sure what to do or try next, i did try adding key in database but think smtp has changed to much for it

thanks

**Comment by ppeyman** [comment]
> 

same us i have try all solution but doesn't connect show below error : 

`Cannot connect to server. Please try again later.`

**Comment by cbtrident** [comment]
i went back to ubuntu and it working now i will see if rocky does it again but am working now

**Comment by arun-magento** [comment]
when i add brand logo in edit attribute page same error, i am using shopbybrand extension
